ambry-api/src/test/java/com.github.ambry/account/AccountContainerTest.java

```diff
@@ -0,0 +1,574 @@
+      assertEquals("Wrong behavior in serialize and then deserialize", account, Account.fromJson(account.toJson()));
+      assertEquals("Failed to compare account to a reference account", Account.fromJson(accountJsonLike), account);
+    }
+    if (compareContainer) {
+      Collection<Container> containersFromAccount = account.getAllContainers();
+      assertEquals(2, containersFromAccount.size());
+      for (int i = 0; i < 2; i++) {
+        assertContainer(account.getContainerById(refContainerIds.get(i)), i);
+        assertContainer(account.getContainerByName(refContainerNames.get(i)), i);
+      }
+    }
+  }
+
+  /**
+   * Asserts a {@link Container} against the reference account for every internal field, {@link Container#toJson()}
+   * method, and also asserts the same object after serialize and then deserialize.
+   * @param container The {@link Container} to assert.
+   * @param index The index in the reference container list to assert against.
+   * @throws JSONException
+   */
+  private void assertContainer(Container container, int index) throws JSONException {
+    assertEquals((short) refContainerIds.get(index), container.getId());
+    assertEquals(refContainerNames.get(index), container.getName());
+    assertEquals(refContainerStatuses.get(index), container.getStatus());
+    assertEquals(refContainerDescriptions.get(index), container.getDescription());
+    assertEquals(refContainerPrivacies.get(index), container.getIsPrivate());
+    assertEquals(refAccountId, container.getParentAccountId());
+    assertEquals(containerJsonLikeList.get(index).toString(), container.toJson().toString());
+    assertEquals(Container.fromJson(containerJsonLikeList.get(index)), container);
+  }
+
+  /**
+   * Asserts that create an {@link Account} fails and throw an exception as expected, when supplying an invalid
+   * list of {@link Container}s.
+   * @param containers A list of invalid {@link Container}s.
+   * @param exceptionClass The class of expected exception.
+   */
+  private void assertCreateAccountWithBadContainers(List<Container> containers, Class<?> exceptionClass) {
+    try {
+      new Account(refAccountId, refAccountName, refAccountStatus, containers);
+      fail("should have thrown");
+    } catch (Exception e) {
+      System.out.println(e.getMessage());
+      assertEquals("Wrong exception", exceptionClass, e.getClass());
+    }
+  }
+
+  /**
+   * Asserts that create an {@link Account} fails and throw an exception as expected, when supplying an invalid
+   * metadata in Json.
+   * @param metadata An invalid metadata in Json.
+   * @param exceptionClass The class of expected exception.
+   */
+  private void assertCreatAccountWithBadJsonMetadata(JSONObject metadata, Class<?> exceptionClass) {
+    try {
+      Account.fromJson(metadata);
+      fail("should have thrown");
+    } catch (Exception e) {
+      System.out.println(e.getMessage());
+      assertEquals("Wrong exception", exceptionClass, e.getClass());
```

Comment:
why do we need to print the msg here? You can add the msg in the assertion failure msg.